File: 49-group-anagrams/group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        map<string,vector<string>> ump;
        string alp;
        for(string str: strs){
            alp=str;
            sort(alp.begin(),alp.end());
            ump[alp].push_back(str);
        }
        vector<vector<string>> res;
        for(auto v:ump){
            res.push_back(v.second);
        }
        return res;
    }
};

File: 49-group-anagrams/group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        Map<String,List<String>> anagramsMap = new HashMap<>();
        char[] currArr; String currStr;
        for(String str:strs){
            currArr = str.toCharArray(); Arrays.sort(currArr);
            currStr = new String(currArr);
            List<String> anagrams = anagramsMap.getOrDefault(currStr, new ArrayList<>());
            anagrams.add(str);
            anagramsMap.put(currStr, anagrams);
        }
        List<List<String>> ans = new ArrayList<>(anagramsMap.values());
        return ans;
    }
}